### PR TITLE
feat: Vite integration

### PR DESCRIPTION
**Pull request by prajjwal-devtron**
Vite integration for fe common lib

### CODE REVIEW

File: src/Common/UseWindowSize.tsx
Comment (vivek-devtron) [inline src/Common/UseWindowSize.tsx:1]: Please remove this file it is added in hooks.

File: tsconfig-build.json
Comment (vivek-devtron) [inline tsconfig-build.json:1]: Please format this file

File: tsconfig.node.json
Comment (vivek-devtron) [inline tsconfig.node.json:1]: Please format this file

File: yarn.lock
Comment (vivek-devtron) [inline yarn.lock:1]: Please remove this file as we use npm

File: tsconfig.node.json
Comment (prajjwal-devtron) [reply tsconfig.node.json:1]: done

File: yarn.lock
Comment (prajjwal-devtron) [reply yarn.lock:1]: done

File: tsconfig-build.json
Comment (prajjwal-devtron) [reply tsconfig-build.json:1]: done

File: src/Common/UseWindowSize.tsx
Comment (prajjwal-devtron) [reply src/Common/UseWindowSize.tsx:1]: done

File: tsconfig.node.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+    "compilerOptions": {
+        "composite": true,
+        "skipLibCheck": true,
+        "module": "ESNext",
+        "moduleResolution": "bundler",
```
Comment (eshankvaish) [inline tsconfig.node.json:6]: Argument for '--moduleResolution' option must be: 'node', 'classic', 'node16', 'nodenext'.ts

**Review by eshankvaish — CHANGES_REQUESTED** [review]

**Comment by eshankvaish** [comment]
Pls verify if the node version needs to be bumped for both the libraries

File: tsconfig.node.json (hunk 1 above)
Comment (vivek-devtron) [reply tsconfig.node.json:6]: Ignoring this as not required

File: src/Common/Constants.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
-import { RegistryTypeDetailType } from "./Types"
+import { RegistryTypeDetailType } from './Types'
 
 export const FALLBACK_REQUEST_TIMEOUT = 60000
-export const Host = process.env.REACT_APP_ORCHESTRATOR_ROOT
+export const Host = import.meta.env.REACT_APP_ORCHESTRATOR_ROOT ?? '/orchestrator'
```
Comment (eshankvaish) [inline src/Common/Constants.ts:4]:
```suggestion
export const Host = import.meta.env.VITE_REACT_APP_ORCHESTRATOR_ROOT ?? '/orchestrator'
```

```suggestion
export const Host = window._env_.VITE_REACT_APP_ORCHESTRATOR_ROOT ?? '/orchestrator'
```

File: src/Common/CodeEditor/CodeEditor.tsx [hunk 3]
```diff
@@ -0,0 +1,526 @@
...
+window.MonacoEnvironment = {
+    // @ts-ignore
+    getWorker(workerId, label: string) {
+        switch (label) {
+            case 'editorWorkerService':
+                return new Worker(new URL('monaco-editor/esm/vs/editor/editor.worker.js', import.meta.url))
+            case 'json':
+                return new Worker(new URL('monaco-editor/esm/vs/language/json/json.worker.js', import.meta.url))
+            case 'yaml':
+                return new Worker(new URL('monaco-yaml/yaml.worker.js', import.meta.url))
```
Comment (eshankvaish) [inline src/Common/CodeEditor/CodeEditor.tsx:38]: same here

File: src/Common/Api.ts [hunk 4]
```diff
@@ -74,17 +74,19 @@ const responseMessages = {
 }
 
 function handleLogout() {
-    const continueParam = `${window.location.pathname.replace(process.env.PUBLIC_URL, '')}${window.location.search}`
-    window.location.href = `${window.location.origin}${process.env.PUBLIC_URL}${URLS.LOGIN_SSO}?continue=${continueParam}`
+    const continueParam = `${window.location.pathname.replace(import.meta.env.PUBLIC_URL, '')}${window.location.search}`
+    window.location.href = `${window.location.origin}${import.meta.env.PUBLIC_URL}${
+        URLS.LOGIN_SSO
+    }?continue=${continueParam}`
 }
```
Comment (eshankvaish) [inline src/Common/Api.ts:81]: here as well

**Review by eshankvaish — CHANGES_REQUESTED** [review]

**Review by eshankvaish — APPROVED** [review]

**Review by vivek-devtron — APPROVED** [review]

**Pull request merged**